// @universal-robots/urcap-utils/bin/lib/urservice-helper.js
const fs = require("fs");
const ApiClient = require("./external-urcap-api/ApiClient").default;
const { UrcapsApi } = require("./external-urcap-api");

const apiClient = new ApiClient();
apiClient.basePath = "http://" + getHost() + "/universal-robots/robot-api";
const urcapsApi = new UrcapsApi(apiClient);

class UrcapInstallationError extends Error {
  constructor(message, details, statusCode) {
    super(message);
    this.details = details;
    this.statusCode = statusCode;
  }
}

function getHost() {
  const args = process.argv.slice(2);
  let idx = args.indexOf("--host");
  let host = "localhost";
  if (process.env.DEV_CONTAINER === "true") {
    host = "host.docker.internal";
  }
  if (idx > -1 && args.length >= idx) {
    // User has specified a host
    host = args[idx + 1];
  }
  let port = "80";
  idx = args.indexOf("--port");
  if (idx > -1 && args.length >= idx) {
    port = args[idx + 1];
  }
  return `${host}:${port}`;
}

function checkIfHostReachableAndUrcapExists(vendorId, urcapId) {
  return new Promise((resolve, reject) => {
    urcapsApi.infoUrcap(vendorId, urcapId, (err, _res) => {
      if (err && err.errno === -51) {
        const msg = "Host " + err.address + " unreachable on port " + err.port;
        reject(
          new UrcapInstallationError(
            (message = msg),
            (details = err.code),
            (statusCode = err.code),
          ),
        );
      } else if (err) {
        resolve(false);
      } else {
        resolve(true);
      }
    });
  });
}

async function installOrUpdateUrcap(
  urcapxFile,
  vendorId,
  urcapId,
  forceReinstall = false,
) {
  let doesUrcapExists;
  try {
    doesUrcapExists = await checkIfHostReachableAndUrcapExists(
      vendorId,
      urcapId,
    );
  } catch (err) {
    throw err;
  }

  if (doesUrcapExists && forceReinstall) {
    await deleteUrcap(vendorId, urcapId);
    return await installUrcap(urcapxFile);
  } else if (doesUrcapExists) {
    return await updateUrcap(urcapxFile);
  }
  return await installUrcap(urcapxFile);
}

async function deleteUrcap(vendorId, urcapId) {
  let doesUrcapExists;
  try {
    doesUrcapExists = await checkIfHostReachableAndUrcapExists(
      vendorId,
      urcapId,
    );
  } catch (err) {
    throw err;
  }

  if (!doesUrcapExists) {
    throw new UrcapInstallationError("URCap does not exist");
  }

  return new Promise((resolve, reject) => {
    urcapsApi.deleteGivenUrcap(vendorId, urcapId, (error) => {
      if (error) {
        reject(
          new UrcapInstallationError(
            "Unable to delete URCap",
            error.response.body.message,
            error.status,
          ),
        );
      } else {
        resolve("URCap successfully deleted");
      }
    });
  });
}

async function installUrcap(filePath) {
  const fileStream = fs.createReadStream(filePath);

  return new Promise((resolve, reject) => {
    urcapsApi.installUrcap(fileStream, (error, msg) => {
      if (error) {
        reject(
          new UrcapInstallationError(
            "Unable to install URCap",
            error.response.body.message,
            error.status,
          ),
        );
      } else {
        resolve(JSON.parse(msg.message));
      }
    });
  });
}

async function updateUrcap(filePath) {
  const fileStream = fs.createReadStream(filePath);

  return new Promise((resolve, reject) => {
    urcapsApi.updateUrcap(fileStream, (error, msg) => {
      if (error) {
        reject(
          new UrcapInstallationError(
            "Unable to update URCap",
            error.response.body.message,
            error.status,
          ),
        );
      } else {
        resolve(JSON.parse(msg.details));
      }
    });
  });
}

module.exports = { deleteUrcap, installOrUpdateUrcap };
