// @universal-robots/urcap-utils/bin/lib/manifest-helper.js
const yaml = require("js-yaml");
const fs = require("fs");
const tarHelper = require("./tar-helper");
const Ajv = require("ajv");
const request = require("sync-request");
const path = require("path");
const process = require("process");

class Urcap {
  get containers() {
    return this._containers;
  }

  set containers(value) {
    value ? (this._containers = value) : (this._containers = []);
  }

  set webArchives(archives) {
    archives ? (this._webArchives = archives) : (this._webArchives = []);
  }

  get webArchives() {
    return this._webArchives;
  }

  constructor() {
    this.vendorId = "";
    this.urcapId = "";
    this.version = "";
    this._webArchives = [];
    this._containers = [];
  }
}

function extractURCapFromManifest(manifest) {
  const webArchives = manifest.artifacts.webArchives;
  const containers = manifest.artifacts.containers;

  const result = new Urcap();
  result.vendorId = manifest.metadata.vendorID;
  result.urcapId = manifest.metadata.urcapID;
  result.version = manifest.metadata.version;
  result.webArchives = webArchives;
  result.containers = containers;

  return result;
}

function readManifestFile(manifestFile) {
  const manifest = yaml.load(fs.readFileSync(manifestFile, "utf8"));
  return extractURCapFromManifest(manifest);
}

function readManifestFileFromUrcap(urcapxFile) {
  return tarHelper
    .readTextFileFromTar(urcapxFile, "manifest.yaml")
    .then((text) => extractURCapFromManifest(yaml.load(text, {})));
}

function getManifestFileFromUrcapAsYaml(urcapxFile) {
  return tarHelper
    .readTextFileFromTar(urcapxFile, "manifest.yaml")
    .then((text) => yaml.load(text));
}

function fetchJsonSync(source) {
  if (source.startsWith("http://") || source.startsWith("https://")) {
    // Handle URL synchronously
    try {
      const response = request("GET", source);
      const data = response.getBody("utf8");
      return JSON.parse(data);
    } catch (err) {
      console.log(`Error. Could not parse the schema URL '${source}'`);
      process.exit(1);
    }
  } else {
    // Handle file path synchronously
    // Check if schema file exists
    const filePath = path.resolve(source);
    try {
      const data = fs.readFileSync(filePath, "utf8");
      return JSON.parse(data);
    } catch (err) {
      console.log(`Error. Could not parse the schema-file '${source}'`);
      process.exit(1);
    }
  }
}

/**
 * Validates the yaml content against the json schema
 * @param schema as json
 * @param manifest as yaml
 * @returns {object} validation result object with status, message, and errors
 */
function validate(schema, manifest) {
  try {
    // Remove the $schema property to avoid Draft 2020-12 issues
    const { $schema, ...schemaWithoutDraft } = schema;

    // Create a new Ajv instance with options to handle $refs
    const ajvInstance = new Ajv({
      strict: false,
      allErrors: true,
      validateFormats: false,
      addUsedSchema: false, // Don't automatically add schemas
    });

    // Validate the URCap manifest using schema without $schema
    const validate = ajvInstance.compile(schemaWithoutDraft);
    const valid = validate(manifest);

    if (!valid) {
      return {
        success: false,
        status: "validation_failed",
        message: "Manifest validation failed",
        errors: validate.errors.map((error) => ({
          path: error.instancePath,
          message: error.message,
          schemaPath: error.schemaPath,
          keyword: error.keyword,
          params: error.params.additionalProperty,
        })),
      };
    } else {
      return {
        success: true,
        status: "validation_passed",
        message: "Manifest validation passed!",
        errors: [],
      };
    }
  } catch (error) {
    return {
      success: false,
      status: "validation_error",
      message: `Error during validation: ${error.message}`,
      errors: [
        {
          path: "",
          message: error.message,
          type: "system_error",
        },
      ],
    };
  }
}

module.exports = {
  readManifestFile,
  readManifestFileFromUrcap,
  getManifestFileFromUrcapAsYaml,
  validate,
  fetchJsonSync,
};
