// @universal-robots/urcap-utils/bin/lib/external-urcap-api/api/InternalApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _APIError = _interopRequireDefault(require("../model/APIError"));
var _APIResponse = _interopRequireDefault(require("../model/APIResponse"));
var _HTTPValidationError = _interopRequireDefault(require("../model/HTTPValidationError"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * URCapX System API - Urservice Domain
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
* Internal service.
* @module api/InternalApi
* @version 1.0.0
*/
var InternalApi = exports["default"] = /*#__PURE__*/function () {
  /**
  * Constructs a new InternalApi. 
  * @alias module:api/InternalApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function InternalApi(apiClient) {
    _classCallCheck(this, InternalApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Callback function to receive the result of the deleteGivenUrcap operation.
   * @callback module:api/InternalApi~deleteGivenUrcapCallback
   * @param {String} error Error message, if any.
   * @param {module:model/APIResponse} data The data returned by the service call.
   * @param {String} response The complete HTTP response.
   */

  /**
   * Delete Given Urcap
   * Deletes a URCap
   * @param {String} vendorId 
   * @param {String} urcapId 
   * @param {module:api/InternalApi~deleteGivenUrcapCallback} callback The callback function, accepting three arguments: error, data, response
   * data is of type: {@link module:model/APIResponse}
   */
  return _createClass(InternalApi, [{
    key: "deleteGivenUrcap",
    value: function deleteGivenUrcap(vendorId, urcapId, callback) {
      var postBody = null;
      // verify the required parameter 'vendorId' is set
      if (vendorId === undefined || vendorId === null) {
        throw new Error("Missing the required parameter 'vendorId' when calling deleteGivenUrcap");
      }
      // verify the required parameter 'urcapId' is set
      if (urcapId === undefined || urcapId === null) {
        throw new Error("Missing the required parameter 'urcapId' when calling deleteGivenUrcap");
      }
      var pathParams = {
        'vendor_id': vendorId,
        'urcap_id': urcapId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _APIResponse["default"];
      return this.apiClient.callApi('/urcaps/v1/urcaps/{vendor_id}/{urcap_id}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the infoUrcap operation.
     * @callback module:api/InternalApi~infoUrcapCallback
     * @param {String} error Error message, if any.
     * @param {module:model/APIResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Info Urcap
     * Get information about a URCap
     * @param {String} vendorId 
     * @param {String} urcapId 
     * @param {module:api/InternalApi~infoUrcapCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/APIResponse}
     */
  }, {
    key: "infoUrcap",
    value: function infoUrcap(vendorId, urcapId, callback) {
      var postBody = null;
      // verify the required parameter 'vendorId' is set
      if (vendorId === undefined || vendorId === null) {
        throw new Error("Missing the required parameter 'vendorId' when calling infoUrcap");
      }
      // verify the required parameter 'urcapId' is set
      if (urcapId === undefined || urcapId === null) {
        throw new Error("Missing the required parameter 'urcapId' when calling infoUrcap");
      }
      var pathParams = {
        'vendor_id': vendorId,
        'urcap_id': urcapId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _APIResponse["default"];
      return this.apiClient.callApi('/urcaps/v1/urcaps/{vendor_id}/{urcap_id}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the installUrcap operation.
     * @callback module:api/InternalApi~installUrcapCallback
     * @param {String} error Error message, if any.
     * @param {module:model/APIResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Install Urcap
     * Installs a URCap from a .urcapx file
     * @param {File} urcapxFile 
     * @param {module:api/InternalApi~installUrcapCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/APIResponse}
     */
  }, {
    key: "installUrcap",
    value: function installUrcap(urcapxFile, callback) {
      var postBody = null;
      // verify the required parameter 'urcapxFile' is set
      if (urcapxFile === undefined || urcapxFile === null) {
        throw new Error("Missing the required parameter 'urcapxFile' when calling installUrcap");
      }
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {
        'urcapx_file': urcapxFile
      };
      var authNames = [];
      var contentTypes = ['multipart/form-data'];
      var accepts = ['application/json'];
      var returnType = _APIResponse["default"];
      return this.apiClient.callApi('/urcaps/v1/urcaps/', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the listAllUrcaps operation.
     * @callback module:api/InternalApi~listAllUrcapsCallback
     * @param {String} error Error message, if any.
     * @param {module:model/APIResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * List Urcaps
     * Gets a list of all installed URCaps
     * @param {module:api/InternalApi~listAllUrcapsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/APIResponse}
     */
  }, {
    key: "listAllUrcaps",
    value: function listAllUrcaps(callback) {
      var postBody = null;
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _APIResponse["default"];
      return this.apiClient.callApi('/urcaps/v1/urcaps/', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the updateGivenUrcap operation.
     * @callback module:api/InternalApi~updateGivenUrcapCallback
     * @param {String} error Error message, if any.
     * @param {module:model/APIResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Update Given Urcap
     * Updates a URCap from a .urcapx file
     * @param {String} vendorId 
     * @param {String} urcapId 
     * @param {File} urcapxFile 
     * @param {module:api/InternalApi~updateGivenUrcapCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/APIResponse}
     */
  }, {
    key: "updateGivenUrcap",
    value: function updateGivenUrcap(vendorId, urcapId, urcapxFile, callback) {
      var postBody = null;
      // verify the required parameter 'vendorId' is set
      if (vendorId === undefined || vendorId === null) {
        throw new Error("Missing the required parameter 'vendorId' when calling updateGivenUrcap");
      }
      // verify the required parameter 'urcapId' is set
      if (urcapId === undefined || urcapId === null) {
        throw new Error("Missing the required parameter 'urcapId' when calling updateGivenUrcap");
      }
      // verify the required parameter 'urcapxFile' is set
      if (urcapxFile === undefined || urcapxFile === null) {
        throw new Error("Missing the required parameter 'urcapxFile' when calling updateGivenUrcap");
      }
      var pathParams = {
        'vendor_id': vendorId,
        'urcap_id': urcapId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {
        'urcapx_file': urcapxFile
      };
      var authNames = [];
      var contentTypes = ['multipart/form-data'];
      var accepts = ['application/json'];
      var returnType = _APIResponse["default"];
      return this.apiClient.callApi('/urcaps/v1/urcaps/{vendor_id}/{urcap_id}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the updateUrcap operation.
     * @callback module:api/InternalApi~updateUrcapCallback
     * @param {String} error Error message, if any.
     * @param {module:model/APIResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Update Urcap
     * Updates a URCap from a .urcapx file
     * @param {File} urcapxFile 
     * @param {module:api/InternalApi~updateUrcapCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/APIResponse}
     */
  }, {
    key: "updateUrcap",
    value: function updateUrcap(urcapxFile, callback) {
      var postBody = null;
      // verify the required parameter 'urcapxFile' is set
      if (urcapxFile === undefined || urcapxFile === null) {
        throw new Error("Missing the required parameter 'urcapxFile' when calling updateUrcap");
      }
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {
        'urcapx_file': urcapxFile
      };
      var authNames = [];
      var contentTypes = ['multipart/form-data'];
      var accepts = ['application/json'];
      var returnType = _APIResponse["default"];
      return this.apiClient.callApi('/urcaps/v1/urcaps/', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);
}();
