// @universal-robots/ui-models/fesm2022/universal-robots-ui-models.mjs
const DialogSize = {
    TAKEOVER: 'TAKEOVER',
    XXL: 'XXL',
    XL: 'XL',
    LARGE: 'LARGE',
    DEFAULT: 'DEFAULT',
    SMALL: 'SMALL',
    XSMALL: 'XSMALL',
};
const CloseReason = {
    CANCELLED: 'CANCELLED',
    DESTROYED: 'DESTROYED',
    CONFIRMED: 'CONFIRMED',
    UNKNOWN: 'UNKNOWN',
};
var SafetyModeError;
(function (SafetyModeError) {
    SafetyModeError["NORMAL"] = "NORMAL";
    SafetyModeError["REDUCED"] = "REDUCED";
    SafetyModeError["PROTECTIVE_STOP"] = "PROTECTIVE_STOP";
    SafetyModeError["RECOVERY"] = "RECOVERY";
    SafetyModeError["SAFEGUARD_STOP"] = "SAFEGUARD_STOP";
    SafetyModeError["SYSTEM_EMERGENCY_STOP"] = "SYSTEM_EMERGENCY_STOP";
    SafetyModeError["ROBOT_EMERGENCY_STOP"] = "ROBOT_EMERGENCY_STOP";
    SafetyModeError["VIOLATION"] = "VIOLATION";
    SafetyModeError["FAULT"] = "FAULT";
    SafetyModeError["VALIDATE_JOINT_ID"] = "VALIDATE_JOINT_ID";
    SafetyModeError["UNDEFINED_SAFETY_MODE"] = "UNDEFINED_SAFETY_MODE";
    SafetyModeError["AUTOMATIC_MODE_SAFEGUARD_STOP"] = "AUTOMATIC_MODE_SAFEGUARD_STOP";
    SafetyModeError["SYSTEM_THREE_POSITION_ENABLING_STOP"] = "SYSTEM_THREE_POSITION_ENABLING_STOP";
    SafetyModeError["TP_THREE_POSITION_ENABLING_STOP"] = "TP_THREE_POSITION_ENABLING_STOP";
    SafetyModeError["IMMI_EMERGENCY_STOP"] = "IMMI_EMERGENCY_STOP";
    SafetyModeError["IMMI_SAFEGUARD_STOP"] = "IMMI_SAFEGUARD_STOP";
    SafetyModeError["PROFISAFE_WAITING_FOR_PARAMETERS"] = "PROFISAFE_WAITING_FOR_PARAMETERS";
    SafetyModeError["PROFISAFE_AUTOMATIC_MODE_SAFEGUARD_STOP"] = "PROFISAFE_AUTOMATIC_MODE_SAFEGUARD_STOP";
    SafetyModeError["PROFISAFE_SAFEGUARD_STOP"] = "PROFISAFE_SAFEGUARD_STOP";
    SafetyModeError["PROFISAFE_EMERGENCY_STOP"] = "PROFISAFE_EMERGENCY_STOP";
    SafetyModeError["BRAKE_PIN_FAILURE"] = "BRAKE_PIN_FAILURE";
    SafetyModeError["SAFETY_API_SAFEGUARD_STOP"] = "SAFETY_API_SAFEGUARD_STOP";
    SafetyModeError["SAFETY_PLANES_SAFEGUARD_STOP"] = "SAFETY_PLANES_SAFEGUARD_STOP";
})(SafetyModeError || (SafetyModeError = {}));
const DIALOG_KEYS = {
    LOCALSTORAGE: 'ur-portal-service',
    OPEN_EVENT: 'portalservice:open',
    CLOSE_EVENT: 'portalservice:close',
    UPDATE_EVENT: 'portalservice:update',
    CLOSE_BY_ID_EVENT: 'portalservice:close_own',
    CLOSE_BY_ERROR_TYPE_EVENT: 'portalservice:close_by_type',
};

const SelectedInput = {
    VALUE: 'VALUE',
    VARIABLE: 'VARIABLE',
    EXPRESSION: 'EXPRESSION',
};
const TAB_INPUT_VALUE_TYPES = [
    'string',
    'integer',
    'float',
    'boolean',
    'pose',
    'matrix',
    'array',
    'waypoint',
    'grid',
    'timer',
    'profile',
    'frame',
];

const getAllowedKeys = () => {
    const allowedKeys = {
        dot: '.',
        arrowDown: "ArrowDown" /* Key.ArrowDown */,
        arrowLeft: "ArrowLeft" /* Key.ArrowLeft */,
        arrowRight: "ArrowRight" /* Key.ArrowRight */,
        arrowUp: "ArrowUp" /* Key.ArrowUp */,
        home: "Home" /* Key.Home */,
        end: "End" /* Key.End */,
        dash: '-',
        plus: '+',
        tab: "Tab" /* Key.Tab */,
        backspace: "Backspace" /* Key.Backspace */,
        del: "Delete" /* Key.Delete */,
    };
    return [...Object.values(allowedKeys)];
};
var urField_model = {};

/**
 * Generated bundle index. Do not edit.
 */

export { CloseReason, DIALOG_KEYS, DialogSize, SafetyModeError, SelectedInput, TAB_INPUT_VALUE_TYPES, getAllowedKeys };
