// @universal-robots/contribution-api/fesm2022/universal-robots-contribution-api-angular.mjs
import * as i0 from '@angular/core';
import { Pipe, EventEmitter, Component, Output, Input } from '@angular/core';
import { valueConverter, valueRawConverter, convertValueToUnit } from '@universal-robots/contribution-api';
import { FixedPointNumber } from '@universal-robots/utilities-units';
import { CloseReason } from '@universal-robots/ui-models';

/**
 * Formats a Value object to a string converting it to the specified unit with the unit's symbol appended
 * <p>The pipe requires one additional argument, the unit to convert into see {@link transform} for more information</p>
 * @example
 * ```html
 * <span>{{ { value: 1, unit: 'm/s' } | displayvalue: 'mm/s' }}</span>
 * <!-- outputs: <span>1000 mm/s</span> -->
 * ```
 * @param value The value object to be formatted
 * @param to The unit to convert into, must be compatible with the value object's unit
 * @returns The formatted value with the specified unit's symbol appended
 * @see DisplayValueRawPipe
 */
class DisplayValuePipe {
    /**
     * Transform the value object into a formatted string converting it to the specified unit with the unit's symbol appended
     * @param value The value object to be formatted
     * @param to The unit to convert into, must be compatible with the value object's unit
     * @returns The formatted value with the unit's symbol appended
     */
    transform(value, to) {
        return valueConverter(value, to);
    }
    static { this.ɵfac = function DisplayValuePipe_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || DisplayValuePipe)(); }; }
    static { this.ɵpipe = /*@__PURE__*/ i0.ɵɵdefinePipe({ name: "displayvalue", type: DisplayValuePipe, pure: true }); }
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(DisplayValuePipe, [{
        type: Pipe,
        args: [{
                name: 'displayvalue',
                standalone: true,
            }]
    }], null, null); })();
/**
 * Formats a value object to a string converting it to the specified unit without any unit symbols or label
 * <p>The pipe requires one additional argument, the unit to convert into see {@link transform} for more information</p>
 * @example
 * ```html
 * <span>{{ { value: 1, unit: 'm/s' } | displayvalueraw: 'mm/s' }}</span>
 * <!-- outputs: <span>1000</span> -->
 * ```
 * @see DisplayValuePipe
 */
class DisplayValueRawPipe {
    /**
     * Transform the value object into a formatted string converting it to the specified unit without any unit symbols or label
     * @param value The value object to be formatted
     * @param to The unit to convert into, must be compatible with the value object's unit
     * @returns The formatted value without any unit symbols or label
     */
    transform(value, to) {
        return valueRawConverter(value, to);
    }
    static { this.ɵfac = function DisplayValueRawPipe_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || DisplayValueRawPipe)(); }; }
    static { this.ɵpipe = /*@__PURE__*/ i0.ɵɵdefinePipe({ name: "displayvalueraw", type: DisplayValueRawPipe, pure: true }); }
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(DisplayValueRawPipe, [{
        type: Pipe,
        args: [{
                name: 'displayvalueraw',
                standalone: true,
            }]
    }], null, null); })();

class ValueFormatterPipe {
    /**
     * @param value {Value} The value object to get fixed decimal places
     * @param to {Unit} The unit to convert into, must be compatible with the value object's unit
     * @returns The number fixed to the precision specified by the unit
     */
    transform(value, to) {
        const convertedValue = convertValueToUnit(value, to);
        return new FixedPointNumber(convertedValue.value, to).getFixedValue();
    }
    static { this.ɵfac = function ValueFormatterPipe_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || ValueFormatterPipe)(); }; }
    static { this.ɵpipe = /*@__PURE__*/ i0.ɵɵdefinePipe({ name: "valueFormatter", type: ValueFormatterPipe, pure: true }); }
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(ValueFormatterPipe, [{
        type: Pipe,
        args: [{
                name: 'valueFormatter',
                standalone: true,
            }]
    }], null, null); })();

class NumberPrecisionPipe {
    /**
     * @param value The value object to get fixed decimal places
     * @param precision {number, Unit} the precision to round the value to or the unit to convert into the buildin significat digits
     * @returns The number fixed to the specified precision
     */
    transform(value, precision) {
        if (isNaN(value)) {
            return value;
        }
        // Unit guard
        if (typeof precision === 'object' && 'significantDigits' in precision) {
            return Number.parseFloat(value.toFixed(precision.significantDigits));
        }
        return typeof precision === 'number' && !isNaN(precision) ? Number.parseFloat(value.toFixed(precision)) : undefined;
    }
    static { this.ɵfac = function NumberPrecisionPipe_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || NumberPrecisionPipe)(); }; }
    static { this.ɵpipe = /*@__PURE__*/ i0.ɵɵdefinePipe({ name: "precision", type: NumberPrecisionPipe, pure: true }); }
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(NumberPrecisionPipe, [{
        type: Pipe,
        args: [{
                name: 'precision',
                standalone: true,
            }]
    }], null, null); })();

/**
 * Abstract Angular component for extension by Dialogs that implement a custom footer.
 *
 * Gives access to the input data and the presenter API.
 *
 * Provides methods to cancel or confirm the dialog with a reason and return data.
 *
 * For dialogs that do not require a custom footer, use {@link WebComponentDialogComponent}.
 *
 * @see WebComponentDialogComponent
 */
class CustomFooterDialogComponent {
    constructor() {
        /**
         * Closes the dialog.
         *
         * The event contains the reason the dialog was closed and the output data.
         */
        this.closeDialog = new EventEmitter();
    }
    /**
     * Closes the dialog with a reason of {@link CloseReason.CANCELLED} and without any return data.
     * @see CloseDialog
     */
    cancelDialog() {
        this.closeDialog.emit({
            reason: CloseReason.CANCELLED,
        });
    }
    /**
     * Closes the dialog with a reason of {@link CloseReason.CONFIRMED} and with provided returnData as the {@link CloseDialog#returnData}.
     * @param returnData The data to include as the return data in the {@link CloseDialog}
     * @see CloseDialog
     */
    confirmDialog(returnData) {
        this.closeDialog.emit({
            reason: CloseReason.CONFIRMED,
            returnData: returnData,
        });
    }
    static { this.ɵfac = function CustomFooterDialogComponent_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || CustomFooterDialogComponent)(); }; }
    static { this.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: CustomFooterDialogComponent, selectors: [["ng-component"]], inputs: { inputData: "inputData", presenterApi: "presenterApi" }, outputs: { closeDialog: "closeDialog" }, standalone: false, decls: 0, vars: 0, template: function CustomFooterDialogComponent_Template(rf, ctx) { }, encapsulation: 2 }); }
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(CustomFooterDialogComponent, [{
        type: Component,
        args: [{
                template: '',
                standalone: false,
            }]
    }], null, { inputData: [{
            type: Input,
            args: [{ required: true }]
        }], presenterApi: [{
            type: Input,
            args: [{ required: true }]
        }], closeDialog: [{
            type: Output
        }] }); })();
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassDebugInfo(CustomFooterDialogComponent, { className: "CustomFooterDialogComponent", filePath: "lib/components/custom-footer-dialog.component.ts", lineNumber: 20 }); })();

/**
 * Generated bundle index. Do not edit.
 */

export { CustomFooterDialogComponent, DisplayValuePipe, DisplayValueRawPipe, NumberPrecisionPipe, ValueFormatterPipe };
